packages/cloud_firestore/android/src/main/java/io/flutter/plugins/firebase/cloudfirestore/CloudFirestorePlugin.java

```diff
@@ -419,7 +423,7 @@ public void notImplemented() {
                       // Wait till transaction is complete.
                       try {
                         String timeoutKey = "transactionTimeout";
-                        long timeout = ((Number) arguments.get(timeoutKey)).longValue();
+                        Long timeout = call.argument(timeoutKey);
```

Comment:
I found this surprising, I would expect that you'd need to call `longValue` here if an `int` was passed from Dart. Are you sure?

Reply:
Well, it should work.
If you look [here](https://flutter.dev/docs/development/platform-integration/platform-channels#platform-channel-data-types-support-and-codecs) you can see that a dart `int` can fit a java `Long`.

Reply:
Ohh, I tested. It broke. Moving to Integer solves.